File: web-app/src/lib/features/features/client-event.ts
import type { Feature } from '../types';

export const clientEventFeatures = {
  'client-event-log': {
    id: 'client-event-log',
    title: 'Log Client Events',
    description: 'Receives and records client-side events from the frontend for server-side processing.',
    rpcIds: ['client-event:log'],
    componentPaths: [],
    testIds: [],
    status: 'experimental',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/client_event.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// ClientEventLog describes the log-client-events RPC.
var ClientEventLog = featureregistry.Feature{
	ID:          "client-event-log",
	Title:       "Log Client Events",
	Description: "Receives and records client-side events from the frontend for server-side processing.",
	RPCIDs:      []string{"client-event:log"},
	Status:      featureregistry.StatusExperimental,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(ClientEventLog)
}
